components/tikv_util/src/file.rs

```diff
@@ -46,6 +46,13 @@ pub fn create_dir_if_not_exist<P: AsRef<Path>>(dir: P) -> io::Result<bool> {
     }
 }
 
+/// Call fsync on file or directory by its path
+pub fn fsync_by_path<P: AsRef<Path>>(path: P) -> io::Result<()> {
```

Comment:
Please add a unit test for this fn.

Reply:
s/fsync_by_path/sync_dir

Reply:
However this function can also be used to call fsync on a normal file

Reply:
Currently no other where use it, please keep the semantics clear.

Reply:
Done